heartbeat/monitors/monitor.go

```diff
@@ -246,6 +308,9 @@ func (m *Monitor) Stop() {
 	m.internalsMtx.Lock()
 	defer m.internalsMtx.Unlock()
 
+	// Free up the monitor ID for reuse
+	uniqueMonitorIDs.Delete(m.id)
```

Comment:
Should it only be freed up after all jobs and tasks are stopped? Perhaps use defer?